test/gurka/test_instructions_internal_intersection.cc

```diff
@@ -0,0 +1,117 @@
+         {"EF", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"FG", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"GH", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"IG", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}},
+         {"GB", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"BJ", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"KC", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"CF", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"FL", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}}};
+
+    const auto layout =
+        gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+
+    map =
+        gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_instructions_internal_intersection",
+                          {{"mjolnir.admin",
+                            {VALHALLA_SOURCE_DIR "test/data/netherlands_admin.sqlite"}}});
+  }
+};
+
+gurka::map InstructionsInternalIntersection::map = {};
+
+///////////////////////////////////////////////////////////////////////////////
+// Left turn
+TEST_F(InstructionsInternalIntersection, BrokenLandParkway_TurnLeft_PatuxentWoodsDrive) {
+  auto result = gurka::route(map, "A", "L", "auto");
+
+  // Verify path
+  gurka::assert::raw::expect_path(result, {"Broken Land Parkway", "Broken Land Parkway",
+                                           "Snowden River Parkway", "Patuxent Woods Drive"});
+
+  // Verify maneuver types
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  int maneuver_index = 0;
+
+  // Verify start distance includes internal edge
+  gurka::assert::raw::expect_instructions_at_maneuver_index(
+      result, maneuver_index, "Drive north on Broken Land Parkway.", "",
+      "Drive north on Broken Land Parkway. Then Turn left onto Patuxent Woods Drive.",
+      "Continue for 70 meters.");
+
+  // Verify the turn left onto Patuxent Woods Drive instructions
+  gurka::assert::raw::expect_instructions_at_maneuver_index(
+      result, ++maneuver_index, "Turn left onto Patuxent Woods Drive.",
+      "Turn left onto Patuxent Woods Drive.",
+      "Turn left onto Patuxent Woods Drive. Then You will arrive at your destination.",
+      "Continue for 40 meters.");
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Left u-turn
+TEST_F(InstructionsInternalIntersection, BrokenLandParkway_UturnLeft_BrokenLandParkway) {
```

Comment:
Are there other tests that would exercise a route that does not take an internal intersection? Just so we are sure we haven't changed anything for those scenarios? Rest LGTM.

Reply:
covered scenarios with RAD - will see about adding more into gurka